templates/types/streaming/nextjs/app/components/ui/document-preview.tsx

```diff
@@ -14,24 +17,27 @@ import {
 } from "./drawer";
 import { cn } from "./lib/utils";
 
-export interface UploadCsvPreviewProps {
-  csv: CsvFile;
+export interface DocumentPreviewProps {
+  file: DocumentFile;
   onRemove?: () => void;
 }
 
-export default function UploadCsvPreview(props: UploadCsvPreviewProps) {
-  const { filename, filesize, content } = props.csv;
+export function DocumentPreview(props: DocumentPreviewProps) {
+  const { filename, filesize, content, filetype } = props.file;
+  const docContent = Array.isArray(content)
```

Comment:
TODO: this would just display IDs

workaround: 
1. disable previewing private documents for now OR
2. store uploaded files in `/data/` folder and use open file via file-server